pkg/kubelet/cm/devicemanager/pod_devices.go

```diff
@@ -344,18 +344,20 @@ func (pdev *podDevices) getContainerDevices(podUID, contName string) ResourceDev
 		devicePluginMap := make(map[string]pluginapi.Device)
 		for numaid, devlist := range allocateInfo.deviceIds {
 			for _, devId := range devlist {
-				NUMANodes := []*pluginapi.NUMANode{{ID: numaid}}
-				if pDev, ok := devicePluginMap[devId]; ok && pDev.Topology != nil {
-					if nodes := pDev.Topology.GetNodes(); nodes != nil {
-						NUMANodes = append(NUMANodes, nodes...)
+				var topology *pluginapi.TopologyInfo
+				if numaid != nodeWithoutTopology {
```

Comment:
I think rather than skipping devices that don't have topology we should expose NUMA ID as -1 to the podresource API client that there are devices without topology and let the client decide what it would like to do with this information. This might need some additional documentation but I think just simply skipping the devices is not the best approach because it gives the impression that devices with the Topology are the only allocated devices which might lead to incorrect accounting of resources if that is the goal of the client.

Reply:
device is not skipped, it will be there, but with empty Topology field.

Reply:
Ah right! Sorry I misread. Looks good!